codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelMethodOverrides.java

```diff
@@ -55,21 +62,25 @@ public MethodSpec equalsMethod(ShapeModel shapeModel) {
             methodBuilder.addStatement("$T other = ($T) obj", className, className);
         }
 
-        shapeModel.getNonStreamingMembers().forEach(m -> {
-            String getterName = m.getFluentGetterMethodName();
-            methodBuilder.beginControlFlow("if (other.$N() == null ^ this.$N() == null)", getterName, getterName)
-                         .addStatement("return false")
-                         .endControlFlow()
-
-                         .beginControlFlow("if (other.$N() != null && !other.$N().equals(this.$N()))", getterName, getterName,
-                                           getterName)
-                         .addStatement("return false")
-                         .endControlFlow();
-        });
-
-        methodBuilder.addStatement("return true");
+        final List<MemberModel> memberModels = shapeModel.getNonStreamingMembers();
+        final CodeBlock.Builder memberEqualsStmt = CodeBlock.builder();
+        if (memberModels.isEmpty()) {
+            memberEqualsStmt.addStatement("return true");
+        } else {
+            final BiConsumer<String, MemberModel> adder = (sep, m) -> {
+                final String getterName = m.getFluentGetterMethodName();
+                memberEqualsStmt.add(sep)
+                        .add(" ")
+                        .add("$T.equals($N(), other.$N())", Objects.class, getterName, getterName);
+            };
+            memberModels.stream().findFirst().ifPresent(m -> adder.accept("return", m));
+            memberModels.stream()
+                    .skip(1)
+                    .forEach(m -> adder.accept("&&", m));
+            memberEqualsStmt.add(";");
```

Comment:
Does the [PoetCollectors.toDelimitedCodeBlock](https://github.com/aws/aws-sdk-java-v2/blob/master/codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetCollectors.java#L41) make this a bit easier to implement?

Reply:
Nice! Didn't know that was there. Done